src/main/java/org/rumbledb/runtime/operational/InstanceOfIterator.java

```diff
@@ -68,10 +68,12 @@ public Item next() {
                 if (isInvalidArity(items.size()))
                     return ItemFactory.getInstance().createBooleanItem(false);
 
-                ItemType itemType = this.sequenceType.getItemType();
-                for (Item item : items) {
-                    if (!item.isTypeOf(itemType)) {
-                        return ItemFactory.getInstance().createBooleanItem(false);
+                if (!this.sequenceType.isEmptySequence()) {
+                    ItemType itemType = this.sequenceType.getItemType();
+                    for (Item item : items) {
+                        if (!item.isTypeOf(itemType)) {
+                            return ItemFactory.getInstance().createBooleanItem(false);
+                        }
```

Comment:
From a correctness perspective this is absolutely acceptable. However, I get a little bit of spaghetti code feeling when I look at how this code block and isInvalidArity() function works together to cover various combinations. What do you think of this ?

Reply:
This is fixed.